### PR TITLE
build: run mdc prototype tests with bazel

### PR DESCRIPTION
**Pull request by devversion**
Sets up the MDC prototype entry-points to run tests with Bazel. This
allows us to run tests for these components against Ivy.

### CODE REVIEW

File: src/material-experimental/mdc-checkbox/checkbox.ts [hunk 1]
```diff
@@ -156,6 +158,7 @@ export class MatCheckbox implements AfterViewInit, OnDestroy, ControlValueAccess
   }
   set required(required) {
     this._required = coerceBooleanProperty(required);
+    this._changeDetectorRef.markForCheck();
```
Comment (devversion) [inline src/material-experimental/mdc-checkbox/checkbox.ts]: Not entirely sure what we want to do in general about these. I added a note for this to the meeting so we can chat about it.

Comment (mmalerba) [reply src/material-experimental/mdc-checkbox/checkbox.ts]: I think this makes sense, @crisbeto mentioned a while ago that I should add them, but I forgot

**Comment by jelbourn** [comment]
@devversion what's with the ivy task failure?

**Comment by devversion** [comment]
@jelbourn That's actually the flake we've been chatting about in the components channel. It also happens on `master` and is not specific to Ivy. I'll just spend some time looking into this now. It's quite confusing.

**Comment by devversion** [comment]
Also just realized. Should we keep the history of the commits for this PR as well? (to retain the commit message context)

**Pull request merged**